app/scripts/lib/setupSentry.js

```diff
@@ -118,16 +118,20 @@ export default function setupSentry({ release, getState }) {
    * @returns `true` if MetaMask's state has been initialized, and MetaMetrics
    * is enabled, `false` otherwise.
    */
-  function getMetaMetricsEnabled() {
-    if (getState) {
-      const appState = getState();
-      if (!appState?.store?.metamask?.participateInMetaMetrics) {
-        return false;
-      }
-    } else {
+  async function getMetaMetricsEnabled() {
+    const appState = getState();
+    if (Object.keys(appState) > 0) {
```

Comment:
Will `Object.keys(appState) > 0` always be true because of what `getSentryState` returns?
```
function setupSentryGetStateGlobal(store) {
  global.stateHooks.getSentryState = function () {
    const fullState = store.getState();
    const debugState = maskObject({ metamask: fullState }, SENTRY_STATE);
    return {
      browser: window.navigator.userAgent,
      store: debugState,
      version: platform.getVersion(),
    };
  };
}
```

Reply:
That hook isn't setup until after initialization. The app state should always be popupated with a _real_ representation of the background state if it's non-empty.

The same goes for the UI - the hook isn't setup until after initialization.

Reply:
Yes, exactly